docs/src/pages/guides/migration-v4/migration-v4.md

```diff
@@ -2277,6 +2278,71 @@ As the core components use emotion as their style engine, the props used by emot
   +import { DistributiveOmit } from '@material-ui/types';
   ```
 
+## Migrate theme's `styleOverrides` to emotion
+
+Although your style overrides defined in the theme may partially work, there is an important difference on how the nested elements are styled. The `$` syntax used with JSS will not work with emotion. You need to replace those selectors with a valid class selectors.
+
+### Replace pseudo state class names
+
+```diff
+const theme = createTheme({
+  components: {
+    MuiOutlinedInput: {
+      styleOverrides: {
+        root: {
+-         '&$focused': {
++         '&.Mui-focused': {
+            borderWidth: 1,
+          }
+        }
+      }
+    }
+  }
+});
+```
+
+### Replace nested classes selectors with global class names
+
+```diff
+const theme = createTheme({
+  components: {
+    MuiOutlinedInput: {
+      styleOverrides: {
+        root: {
+-         '& $notchedOutline': {
++         '& .MuiOutlinedInput-notchedOutline': {
+            borderWidth: 1,
+          }
+        }
+      }
+    }
+  }
+});
+```
+
+> Note: For each component we export `[component]Classes` constant that contains all nested classes for that component. You can rely on this instead of hardcoding the classes.
```

Comment:
```suggestion
> Note: For each component we export a `[component]Classes` constant that contains all nested classes for that component. You can rely on this instead of hardcoding the classes.
```